experimental/dds/tree2/src/feature-libraries/sequence-field/utils.ts

```diff
@@ -1037,33 +1037,67 @@ function splitDetachEvent(detachEvent: CellId, length: number): CellId {
 }
 
 export function compareLineages(
-	lineage1: readonly LineageEvent[] | undefined,
-	lineage2: readonly LineageEvent[] | undefined,
+	cell1: CellId,
+	cell2: CellId,
+	metadata: RevisionMetadataSource,
 ): number {
-	if (lineage1 === undefined || lineage2 === undefined) {
+	const [youngerFirst, youngerCell, olderCell] =
+		compareCellAge(cell1, cell2, metadata) < 0 ? [-1, cell1, cell2] : [1, cell2, cell1];
+
+	if (olderCell.lineage === undefined) {
 		return 0;
 	}
 
-	const lineage1Offsets = new Map<RevisionTag, number>();
-	for (const event of lineage1) {
-		lineage1Offsets.set(event.revision, event.offset);
+	const olderFirst = -1 * youngerFirst;
+	const youngerOffsets = new Map<RevisionTag, number>();
+	for (const event of youngerCell.lineage ?? []) {
+		youngerOffsets.set(event.revision, event.offset);
 	}
 
-	for (let i = lineage2.length - 1; i >= 0; i--) {
-		const event2 = lineage2[i];
-		const offset1 = lineage1Offsets.get(event2.revision);
-		if (offset1 !== undefined) {
-			const offset2 = event2.offset;
-			if (offset1 < offset2) {
-				return -1;
-			} else if (offset1 > offset2) {
-				return 1;
+	for (let i = olderCell.lineage.length - 1; i >= 0; i--) {
+		const event = olderCell.lineage[i];
+		const youngerOffset = youngerOffsets.get(event.revision);
+		if (youngerOffset !== undefined) {
+			const olderOffset = event.offset;
+			if (youngerOffset < olderOffset) {
+				return youngerFirst;
+			} else if (youngerOffset > olderOffset) {
+				return olderFirst;
+			}
+		} else {
+			// We've found a cell C that became empty before the younger cell was created.
+			// The younger cell should come before any such cell, so if the older cell comes after C
+			// then we know that the younger cell should come before the older cell.
+			// TODO: Account for the younger cell's tiebreak policy
+			if (event.offset !== 0) {
+				return youngerFirst;
 			}
 		}
 	}
 	return 0;
 }
 
+// Returns 1 if cell1 has more lineage and -1 if cell2 has more lineage.
+// Note that this will return zero if both cells are older than the revision metadata and they both
+// have the same lineage, even if they are not the same cell.
```

Comment:
```suggestion
/**
 * Returns 1 if cell1 has more lineage and -1 if cell2 has more lineage.
 * Note that this will return zero if both cells are older than the revision metadata and they both
 * have the same lineage, even if they are not the same cell.
 */
```